harness/determined/common/experimental/model.py

```diff
@@ -300,9 +305,14 @@ def set_labels(self, labels: List[str]) -> None:
         array of strings. If the model previously had labels, they are replaced.
 
         Arguments:
-            labels (List[string]): All labels to set on the model.
+            labels (List[str]): All labels to set on the model.
         """
-        self.labels = labels
+        if not isinstance(labels, Iterable):
+            raise ValueError(
+                f"set_labels() requires a list of strings as input but got: {labels}"
+            )
+
+        self.labels = list(labels)
```

Comment:
why is this list() call necessary?

Reply:
Previously we'd write whatever you put onto the wire.

I added a check, but not a check to make sure it was actually a `list`, just that we could iterate through it (which is more pythonic).

But this code actually caches the value to the object.  So 1) it shouldn't be the same list object, and 2) the value on the object should actually be a list, even if the user passes a tuple, or the object's typing will be wrong.